File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int ans=0;
        int prev=0;

        for(int i=0;i<bank.length;i++){
            int cur=0;
            for(char ch:bank[i].toCharArray()){
                if(ch=='1') cur++;
            }
            if(cur>0){
                ans+=prev*cur;
                prev=cur;
            }
        }

        return ans;
    }
}

File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int n=bank.size();
        int cnt=0;
        int prev=0;

        for(int i=0;i<n;i++){
            int cur=0;
            for(int j=0;j<bank[i].size();j++){
                if(bank[i][j]=='1') cur++;
            }
            if(cur>0){
                cnt+=prev*cur;
                prev=cur;
            }
        }

        return cnt;
    }
};